### PR TITLE
Add `unregister_system` command

### LINKED ISSUE #16266: Add unregister_system for Commands
## What problem does this solve or what need does it fill?

`Commands` currently has a `register_system` method, but no way to unregister systems. Unregistration is very useful in on_remove hooks.

## What solution would you like?

Add an `unregister_system` method to `Commands`.

## What alternative(s) have you considered?

There are several alternatives, such as writing a custom command for this, but it would be cleaner (and more symmetrical) if this was provided on Commands directly.

### PR DESCRIPTION
**Pull request by grind086**
# Objective

Fixes #16266 

## Solution

Added an `UnregisterSystem` command struct and `Commands::unregister_system`. Also renamed `World::remove_system` and `World::remove_system_cached` to `World::unregister_*`

## Testing

It's a fairly simple change, but I tested locally to ensure it actually works.

### CODE REVIEW

**Comment by BenjaminBrienen** [comment]
Can you add a unit test?

**Review by BenjaminBrienen — APPROVED** [review]

**Comment by grind086** [comment]
Are there any existing tests to use as an example? I couldn't find any for the related commands other than one doctest example for `Commands::run_system`, but I could just be missing them if they're somewhere other than `system::commands::tests` or `system::system_registry::tests`. If not I can come up with something.

**Comment by BenjaminBrienen** [comment]
If there aren't any tests for `register_system`, don't worry about it. We can open a separate issue for adding tests there. (Unless you feel like doing it here)

**Comment by hymm** [comment]
I feel like this shouldn't be called remove_system. That name feels like the opposite of `add_systems`, so I would expect it to work with Schedules.

**Comment by grind086** [comment]
I could change it to `unregister_system` like the original issue suggested if that's preferable. I went with the current name because it's consistent with `World::remove_system`, which is the non-command equivalent.

**Comment by alice-i-cecile** [comment]
Both this and the method it calls should be named `unregister_system`. Once that's done, I'm happy to approve and merge.

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 1]
```diff
@@ -890,6 +891,17 @@ impl<'w, 's> Commands<'w, 's> {
         SystemId::from_entity(entity)
     }
 
+    /// Removes a system previously registered with [`Commands::register_system`] or [`World::register_system`].
+    ///
+    /// See [`World::remove_system`] for more information.
+    pub fn remove_system<I, O>(&mut self, system_id: SystemId<I, O>)
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/commands/mod.rs:897]:
```suggestion
    pub fn unregister_system<I, O>(&mut self, system_id: SystemId<I, O>)
```

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 2]
```diff
@@ -890,6 +891,17 @@ impl<'w, 's> Commands<'w, 's> {
         SystemId::from_entity(entity)
     }
 
+    /// Removes a system previously registered with [`Commands::register_system`] or [`World::register_system`].
+    ///
+    /// See [`World::remove_system`] for more information.
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/commands/mod.rs:896]:
```suggestion
    /// See [`World::unregister_system`] for more information.
```

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 3]
```diff
@@ -890,6 +891,17 @@ impl<'w, 's> Commands<'w, 's> {
...
+    pub fn remove_system<I, O>(&mut self, system_id: SystemId<I, O>)
+    where
+        I: SystemInput + Send + 'static,
+        O: Send + 'static,
+    {
+        self.queue(RemoveSystem::new(system_id));
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/commands/mod.rs:902]:
```suggestion
        self.queue(UnregisterSystem::new(system_id));
```

File: crates/bevy_ecs/src/system/system_registry.rs [hunk 4]
```diff
@@ -544,6 +544,32 @@ where
     }
 }
 
+/// The [`Command`] type for removing one shot systems from [`Commands`](crate::system::Commands).
+pub struct RemoveSystem<I: SystemInput + 'static, O: 'static> {
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/system_registry.rs:548]:
```suggestion
pub struct UnregisterSystem<I: SystemInput + 'static, O: 'static> {
```

File: crates/bevy_ecs/src/system/system_registry.rs [hunk 5]
```diff
@@ -544,6 +544,32 @@ where
     }
 }
 
+/// The [`Command`] type for removing one shot systems from [`Commands`](crate::system::Commands).
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/system_registry.rs:547]:
```suggestion
/// The [`Command`] type for unregistering one-shot systems from [`Commands`](crate::system::Commands).
```

File: crates/bevy_ecs/src/system/system_registry.rs [hunk 6]
```diff
@@ -544,6 +544,32 @@ where
...
+pub struct RemoveSystem<I: SystemInput + 'static, O: 'static> {
+    system_id: SystemId<I, O>,
+}
+
+impl<I, O> RemoveSystem<I, O>
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/system_registry.rs:552]:
```suggestion
impl<I, O> UnregisterSystem<I, O>
```

File: crates/bevy_ecs/src/system/system_registry.rs [hunk 7]
```diff
@@ -544,6 +544,32 @@ where
...
+    I: SystemInput + 'static,
+    O: 'static,
+{
+    /// Creates a new [`Command`] struct, which can be added to [`Commands`](crate::system::Commands).
+    pub fn new(system_id: SystemId<I, O>) -> Self {
+        Self { system_id }
+    }
+}
+
+impl<I, O> Command for RemoveSystem<I, O>
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/system_registry.rs:563]:
```suggestion
impl<I, O> Command for UnregisterSystem<I, O>
```

File: crates/bevy_ecs/src/system/system_registry.rs [hunk 8]
```diff
@@ -544,6 +544,32 @@ where
...
+where
+    I: SystemInput + 'static,
+    O: 'static,
+{
+    fn apply(self, world: &mut World) {
+        let _ = world.remove_system(self.system_id);
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/system_registry.rs:569]:
```suggestion
        let _ = world.unregister_system(self.system_id);
```

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 9]
```diff
@@ -3,7 +3,8 @@ mod parallel_scope;
 use core::{marker::PhantomData, panic::Location};
 
 use super::{
-    Deferred, IntoObserverSystem, IntoSystem, RegisterSystem, Resource, RunSystemCachedWith,
+    Deferred, IntoObserverSystem, IntoSystem, RegisterSystem, RemoveSystem, Resource,
```
Comment (BenjaminBrienen) [inline crates/bevy_ecs/src/system/commands/mod.rs:6]:
```suggestion
    Deferred, IntoObserverSystem, IntoSystem, RegisterSystem, UnregisterSystem, Resource,
```

**Review by BenjaminBrienen — COMMENTED** [review]
I just wanted to try out the suggestion feature lol

File: crates/bevy_ecs/src/system/commands/mod.rs (hunk 9 above)
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/system/commands/mod.rs:6]: This will fail formatting because it's not alphabetized :)

**Comment by grind086** [comment]
I think that's everything, assuming checks pass.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Squeezing into 0.15 because it avoids us shipping the bad `remove_system` method name.

**Pull request merged**